### PR TITLE
Fix "Not Working" issue and allow pressing axis key multiple times to switch modes

### PR DESCRIPTION
**Pull request by McArdellje**
Restructured how the 3 different transform modes are layed out, creating an abstract baseclass for all of them (`BlenderTransformMode`) and moving a lot of generic code that was shared between them into `BlenderManager.cs`
Also I believe fixed the [issue that Poly_0000 was having](https://github.com/RoxDevvv/Unity-Blender-Control-Style/issues/7) by reading key events in `OnDuringSceneGUI` instead of using a custom transform editor
And I added the ability to switch modes by pressing the axis key multiple times, just like in blender
by default it transforms in a screen relative mode, press an axis once and it will transform in global space then press again and it will transform in local space
scaling is identical in world and local space as 1. unity doesn't support skew and 2. that would be weird

### CODE REVIEW

**Comment by McArdellje** [comment]
By "fixed the issue that Poly_0000 was having" I mean I think I fixed the keybinds not working, not whatever GUID / Asset DB related issue they were having, no idea what could cause that

**Comment by RoxDevvv** [comment]
Having an abstract class that allows for the addition of further functionality is a good idea. However, I've noticed a few issues:
1. In Scale mode, the Cancel action no longer works with the right mouse button and the ESC key.
2. Rotation no longer adheres to the axis.
3. The visual black line for scale and rotation, as well as the scale mouse icon, have disappeared.

I will inform you if I encounter further problems.

**Comment by RoxDevvv** [comment]
4 - Scale work in local mode only

**Comment by RoxDevvv** [comment]
5 - in move mode changing axis mode work only by pressing multiple time it is no longer working with unity UI
image

**Comment by McArdellje** [comment]
I'll look into fixing 1 and 2
not sure how to do the black line you mention in 3, but the scale mouse icon I can fix
for 4 I don't think it is possible to make scaling work in any mode other than locally
and 5 I'll add a setting to make it default to the mode set in unity instead

**Comment by RoxDevvv** [comment]
@McArdellje 
For 5, I suggest reflecting the double key press axis change directly in the UI, eliminating the need to configure it as a setting. Refer to the first PR by @JonasWischeropp, which implemented local and global settings; he utilized `Tools.pivotRotation == PivotRotation.Local`.

**Comment by RoxDevvv** [comment]
3 was like that before for rotation and scale ( scale icon and black line )
image

**Comment by McArdellje** [comment]
Right mouse or ESC to cancel works fine for me.
I'm not sure how to get it to draw the black line properly as I did try having something like this in the `DrawSceneGUI` method but it did not seem to display anything
```cs
        Vector3 objectCenter = Vector3.zero;
        if (perObjectData.Count == 1) {
            objectCenter = HandleUtility.WorldToGUIPoint(perObjectData[0].Transform.position);
        } else {
            // average position
            objectCenter = HandleUtility.WorldToGUIPoint(averagePosition);
        }
        Handles.color = Color.black;
        Handles.DrawLine(objectCenter, Event.current.mousePosition);
```

**Comment by RoxDevvv** [comment]
Right mouse and ESC doesn't work only in scale mode 

i will look into drawing the black line

**Comment by McArdellje** [comment]
for me RMB and ESC work in scale mode
as to the line drawing I just tried using `GL` directly and for some reason even this isn't working:
```cs
        GL.PushMatrix();
        GL.LoadPixelMatrix();
        GL.Begin(GL.LINES);
        GL.Color(Color.black);
        GL.Vertex(objectCenter);
        GL.Vertex(Event.current.mousePosition);
        GL.End();
        GL.PopMatrix();
```

**Comment by RoxDevvv** [comment]
My apologies, I meant that ESC and RMB should revert the changes; they are supposed to return the object to its original scale, but currently, they apply the scale. For moving and rotating, it works correctly as it reverts the changes.

**Comment by McArdellje** [comment]
Fixed that issue, still struggling to get the line drawing to work though...

**Comment by RoxDevvv** [comment]
i did that in scale script it kinda worked but the line revered
```
foreach (var data in perObjectData) {
          
            Vector2 cursorPos = Event.current.mousePosition;
            Handles.color = Color.black;
                Handles.DrawLine(data.Transform.position, cursorPos);
        }
 if (BlenderManager.CurrentAxisMode == BlenderManager.AxisMode.Unlocked) {
            return;
        }
```

**Comment by McArdellje** [comment]
So I tested using this code
```cs
        // clear the screen's depth and colour with blue
        GL.Clear(true, true, Color.blue); // same results if false, true (only clear with colour)
        GL.PushMatrix();
        GL.LoadPixelMatrix();
        GL.Begin(GL.LINES);
        GL.Color(Color.black);
        GL.Vertex(Event.current.mousePosition);
        GL.Vertex(objectCenter);
        GL.End();
        GL.PopMatrix();
```
and got this result
image

this is very strange, it makes no sense to me why this would happen, I assumed that I wasn't seeing the line due to it being considered to be "behind" the rest of the scene, but with this the line seems to be "cutting a hole out" using transparency

**Comment by McArdellje** [comment]
`Handles.DrawLine` takes in world space position, so it would probably be easiest at this point to find a way to convert mouse position into world space

**Comment by RoxDevvv** [comment]
i see so all we need is to convert the mouse to world space and we will have it working good

**Comment by McArdellje** [comment]
that adds it, also made it so that when rotating in global or local space the rotation is based on the center of the screen instead of the object's position and the line reflects that

**Comment by McArdellje** [comment]
oh it should be on scale too? give me a second

**Comment by RoxDevvv** [comment]
But in blender the black dashed line always in the center of the object 
i don't think it is good to have black line in center of screen

**Comment by McArdellje** [comment]
oh yea, you're correct, I didn't realise that
though blender has 1 line for all objects instead of multiple and also does rotate around point
I'll make it point towards the average position of all of the objects and have it rotate based on that, so it acts identical to "Individal Origins"

**Comment by RoxDevvv** [comment]
@McArdellje 
Okay, that sounds good. In your way remove the `GetWorldMouse` method from BlenderScale.cs since it appears to be unused.

I've noticed another minor issue: when you select 'Local' in Unity's UI and press 'G', then press any axis, it switches to 'Global'. The expected behavior should be to only switch the axis from the current space to the next space if the key is pressed twice.

**Comment by McArdellje** [comment]
That should do it, if you press the axis twice it will take the current pivotRotation, press it again and it will invert it, then for a 4th time it will go back to "Unlocked" and repeat the cycle
TLDR: it does what you would expect it to

**Comment by RoxDevvv** [comment]
Everything looks good to me. There are some features we could consider adding later, such as how Blender manages multiple objects simultaneously, but that's a topic for another time.

With that refactoring, more developers will find it easier to add features. Thank you so much for your contribution.
